grommet/grommet
# Added script to automatically generate typescript definitions based o…

…n react-desc

#### What does this PR do?

Welcome Typescript to Grommet. Initially we are adding the typescript definitions for the components (automatically generated based on react-desc).

In the near future we also want to add definitions for utils, theme, contexts and maybe even rewrite grommet in typescript? who knows hahaha

#### Where should the reviewer start?

take a look at each component typescript definiton, for example: `src/js/components/Anchor/index.d.ts`.

#### What testing has been done on this PR?

manual

#### How should this be manually tested?

run tests and storybook and make sure nothing breaks.
also use the Grommet components in a project and see if the ts compiler will start complaining about invalid props in grommet components.

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/1512
#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards
